### DISCUSSION TITLE
v3.0.0-rc.1

### DISCUSSION THREAD

**Discussion by jhuleatt** [FirebaseExtended/reactfire#396] [General]
- Mark startWithValue deprecated and alias it to initialData (#310)
- SuspenseWithPerf jsdom compatibility (#324)
- Obey the Rules of Hooks (#367)
- Update Firestore generics (#366)
- Set hasEmitted for initialData (#370)
- Deprecate `AuthCheck` and `ClaimsCheck` components in favor of `useSigninCheck` hook (#368)
- `useSdk`: only check hasValue if suspense mode (#373)
- Update rxfire and rxjs (#372)

<hr /><em>This discussion was created from the release <a href='https://github.com/FirebaseExtended/reactfire/releases/tag/v3.0.0-rc.1'>v3.0.0-rc.1</a>.</em>

**Comment by Smith4170** [comment]
Error on app compile:

Failed to compile
./node_modules/rxfire/database/index.esm.js
Attempted import error: '__spreadArray' is not exported from 'tslib'.

**Reply by jhuleatt** [reply]
@Smith4170 thanks for trying out the new release candidate! I can't reproduce that error with [the example in this repo](https://github.com/FirebaseExtended/reactfire/tree/main/example) or on the [StackBlitz sample](https://stackblitz.com/fork/reactfire-sample). Can you please file an issue with some more detail about your setup (preferably a simple app that reproduces the issue)?

**Reply by epodol** [reply]
~~I'm also having this issue. The repo is available here: https://github.com/epodol/bsclibrary. The only change I have made is update reactfire to v3.0.0-rc.1. I'm continuing to investigate and then I'll open an issue with my findings, or just post here how I solved it.~~

This issue has been solved by updating to `v3.0.0-rc.2`.

**Comment by conrado** [comment]
the example app seems to be throwing some errors on Storage component, for image upload
<img width="1440" alt="2021-07-01_18-21-46" src="https://user-images.githubusercontent.com/77191/124190782-36d96380-da99-11eb-9428-3c7f5a6ac27c.png">

**Reply by jhuleatt** [reply]
Hey @conrado, thanks for trying it out! That `_warmupSubscription undefined` error should have been fixed in the latest build. The example app points at your local build of ReactFire, can you please confirm that you've rebuilt the library, or can you make the `example/package.json` file to point to `v3.0.0-rc.1` with the following change:

```diff
  "dependencies": {
    "display": "^0.1.9",
    "react-app-polyfill": "^1.0.0",
    "react-firebaseui": "^4.2.0",
    "tailwindcss": "^1.9.1",
+  "reactfire": "v3.0.0-rc.1"
  },
  "alias": {
    "react": "../node_modules/react",
    "react-dom": "../node_modules/react-dom",
-    "reactfire": "../."
  },
```

and see if it still happens?

**Reply by conrado** [reply]
issuing a rebuild worked just fine, thank you for the hint

**Comment by conrado** [comment]
I noticed the ConcurrentModeApp component was completely disabled, any chance concurrent mode will make it back?

**Reply by jhuleatt** [reply]
It's still there, it's just that `react@experimental` is now React 18 and throws some errors it didn't used to when you use it in non-concurrent-mode, so I wanted to keep the repo on React 17. If you install `react@experimental` and `react-dom@experimental` and swap out to the `ConcurrentModeApp` in the example, it should all still work:

https://github.com/FirebaseExtended/reactfire/blob/4d1206d615eeaefc790d99b2c5f07cd44219a455/example/index.tsx#L4-L11

**Comment by conrado** [comment]
I'm happy to help with testing and contributing code. Particularly interested in looking into getting the emulators running 100%, though I might need a little guidance and clues

**Reply by jhuleatt** [reply]
Awesome, that would be a huge help! Now is a bit of a tricky time to start contributing because we're working on migrating to [Firebase v9](https://firebase.google.cn/docs/web/learn-more#modular-version) in https://github.com/FirebaseExtended/reactfire/pull/369, which involves some major changes. Once we have a mostly-working build with v9, we'd definitely appreciate lots of testing and contributions!
